### PR TITLE
Let CKernel clean-up. Use unused CSound::Shutdown

### PR DESCRIPTION
**Pull request by Learath2**
Minor cleanup in preparation for the replacement I'll push for #5092.

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [x] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/engine/client/client.cpp [hunk 1]
```diff
@@ -4806,8 +4799,12 @@ int main(int argc, const char **argv)
 		shell_execute(pStorage->GetBinaryPath(PLAT_CLIENT_EXEC, aBuf, sizeof aBuf));
 	}
 
+	pKernel->Shutdown();
 	delete pKernel;
 
+	// shutdown SDL
+	SDL_Quit();
```
Comment (heinrich5991) [inline src/engine/client/client.cpp:4806]: This slightly changes the code paths where `SDL_Quit()` is called, but not for the worse, I think. (Previously, it wasn't called when the `CClient::Run` function would return early.)

File: src/engine/client/sound.cpp [hunk 2]
```diff
@@ -353,7 +353,7 @@ int CSound::Shutdown()
 	SDL_QuitSubSystem(SDL_INIT_AUDIO);
 	free(m_pMixBuffer);
 	m_pMixBuffer = 0;
-	return 0;
+	return;
```
Comment (Robyt3) [inline src/engine/client/sound.cpp:356]:
```suggestion
```

File: src/engine/client/client.cpp (hunk 1 above)
Comment (Learath2) [reply src/engine/client/client.cpp:4806]: I think it's more correct this way, no? An early fail in `CClient::Run` could very well be after `SDL_Init` succeeded.

Comment (heinrich5991) [reply src/engine/client/client.cpp:4806]: Yea, I think it's more correct. :)

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**